x/programs/cmd/simulator/src/lib.rs

```diff
@@ -302,21 +277,26 @@ pub enum ClientError {
 
 #[derive(Error, Debug)]
 pub enum StepError {
-    #[error("Client error {0}")]
+    #[error("Client error: {0}")]
     Client(#[from] ClientError),
     #[error("Serialization / Deserialization error: {0}")]
     Serde(#[from] serde_json::Error),
     #[error("Borsh deserialization error: {0}")]
     BorshDeserialization(#[from] borsh::io::Error),
+    #[error("Program error: {0}")]
+    Program(String),
+    #[error("Int conversion error: {0}")]
+    TryFromInt(#[from] TryFromIntError),
```

Comment:
Where's this being used? This error does not look like the others...

Reply:
In the `TryFrom` implementation to convert a `StepResponse` into a `StepResponseTyped<T>`, that's because there is no implementation of `From<i32>` for `Id`